1.preprocessing_data/scripts/1.evaluate_qc.py

```diff
@@ -136,56 +199,107 @@
 threshold_z = 2
 
 # Identify outlier rows based on Z-scores greater than as to identify whole images with abnormally high saturated pixels
-outliers = df[abs(z_scores) > threshold_z]
+sat_outliers = df[abs(z_scores) > threshold_z]
 
 # Remove any outliers detected in other channels (currently we have concluded that those other channels don't detect artifacts)
-outliers = outliers[(outliers['Channel'] == 'DNA') | (outliers['Channel'] == 'PM')]
+sat_outliers = sat_outliers[(sat_outliers['Channel'] == 'DNA') | (sat_outliers['Channel'] == 'PM')]
+
+# Append saturation outliers to blur outliers, drop any duplicate FOVs and save outliers for QC report
+outliers = pd.concat([sat_outliers, blur_outliers], ignore_index=True).drop_duplicates()
 
 # Save outliers data frame to view in report
 outliers.to_csv("./qc_results/qc_outliers.csv")
 
-print(outliers.shape)
-print(outliers['Channel'].value_counts())
-outliers.sort_values(by='ImageQuality_PercentMaximal')
+print(sat_outliers.shape)
+print(sat_outliers['Channel'].value_counts())
+sat_outliers.sort_values(by='ImageQuality_PercentMaximal', ascending=True).head()
 
 
-# We will use the lowest value in the calculated outliers to determine the threshold where any if one image in a set has a value higher then the whole image set is skipped during IC.
-# 
-# **We will be using 0.28 as the threshold.**
+# In[11]:
 
-# In[10]:
 
+# Calculate the mean and standard deviation
+mean_value = df["ImageQuality_PercentMaximal"].mean()
+std_dev = df["ImageQuality_PercentMaximal"].std()
 
-# Reset the default value to 'inlier'
-df['Outlier_Status'] = 'inlier'
+# Set the threshold multiplier for above and below the mean
+threshold = 2
```

Comment:
If this threshold never changes from 2 why not define the threshold as a constant at the top of this notebook as `THRESHOLD=2`

Reply:
I didn't think about that. I agree this can be set at the start of the notebook, I'll make this change!